OpenRA.Mods.Common/Traits/World/ResourceLayer.cs

```diff
@@ -251,6 +253,8 @@ public ResourceType Harvest(CPos cell)
 			if (c.Type == null)
 				return null;
 
+			--resCells;
```

Comment:
Shouldn't this be inside the `if` below?

Reply:
Looks like indeed that's the reason. My testcase works fine with this moved there.